test/plugins/FakeTimeStampDataGenerator.cpp

```diff
@@ -0,0 +1,190 @@
+      throw dunedaq::dunetrigger::InvalidQueueFatalError(ERS_HERE, get_name(), qi.name, excpt);
+    }
+  }
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting init() method";
+}
+
+void
+FakeTimeStampDataGenerator::get_info(opmonlib::InfoCollector& /*ci*/, int /*level*/)
+{
+}
+
+void
+FakeTimeStampDataGenerator::do_configure(const nlohmann::json& /*obj*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_configure() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_configure() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_start(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_start() method";
+  thread_.start_working_thread();
+  TLOG() << get_name() << " successfully started";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_start() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_stop(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_stop() method";
+  thread_.stop_working_thread();
+  TLOG() << get_name() << " successfully stopped";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_stop() method";
+}
+
+void
+FakeTimeStampDataGenerator::do_scrap(const nlohmann::json& /*args*/)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_scrap() method";
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Exiting do_scrap() method";
+}
+
+ std::vector<triggeralgs::TimeStampedData>
+FakeTimeStampDataGenerator::GetTimestamp()
+{
+  std::vector<triggeralgs::TimeStampedData> tsds;
+  triggeralgs::TimeStampedData tsd{};
+
+  int signaltype = rdm_signaltype(generator);
+
+  auto tsd_start_time = std::chrono::steady_clock::now();
+  tsd.time_stamp = (uint64_t)pd_clock(tsd_start_time.time_since_epoch()).count();
+  tsd.signal_type = signaltype;
+  auto now = std::chrono::steady_clock::now();
+  tsd.counter = (uint32_t)pd_clock(now.time_since_epoch()).count();
+
+  //std::cout << "\033[32mtsd.timestamp: " << tsd.time_stamp << "\033[0m  ";
+  std::cout << "\033[32m" << tsd.time_stamp << ", "<< tsd.signal_type << ", "<< tsd.counter << "\033[0m\n";
+
+  tsds.push_back(tsd);
+  
+  return tsds;
+}
+
+void
+FakeTimeStampDataGenerator::do_work(std::atomic<bool>& running_flag)
+{
+  TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_work() method";
+  size_t generatedCount = 0;
+  size_t sentCount = 0;
+
+  while (running_flag.load()) 
+  {
+    TLOG_DEBUG(TLVL_GENERATION) << get_name() << ": Start of sleep between sends";
+    std::this_thread::sleep_for(std::chrono::nanoseconds(1000000000));
+
+    std::vector<triggeralgs::TimeStampedData> tsds = GetTimestamp();
+
+    if (tsds.size() == 0) 
+    {
+      std::ostringstream oss_prog;
+      oss_prog << "Last TSDs packet has size 0, continuing!";
+      ers::debug(dunedaq::dunetrigger::ProgressUpdate(ERS_HERE, get_name(), oss_prog.str()));
```

Comment:
Use `TLOG_DEBUG` instead of `ers::debug`. You can write `TLOG_DEBUG(level) << "a string";` or `TLOG_DEBUG(level) << some_ers_issue;` whichever you prefer